File: Sources/BizboxNotch/AttendanceAction.swift
enum AttendanceAction {
    case clockIn
    case clockOut

    var title: String {
        switch self {
        case .clockIn:
            return "출근"
        case .clockOut:
            return "퇴근"
        }
    }

    var code: Int {
        switch self {
        case .clockIn:
            return 1
        case .clockOut:
            return 4
        }
    }

    var tabSelector: String {
        switch self {
        case .clockIn:
            return #"li[onclick="fnSetAttOption(1)"]"#
        case .clockOut:
            return #"li[onclick="fnSetAttOption(4)"]"#
        }
    }

    var submitSelector: String {
        switch self {
        case .clockIn:
            return "#attHref1"
        case .clockOut:
            return "#attHref2"
        }
    }

    var resultSelector: String {
        switch self {
        case .clockIn:
            return "#tab1"
        case .clockOut:
            return "#tab2"
        }
    }
}

File: windows/BizboxNotch.Windows/AttendanceAction.cs
namespace BizboxNotch.Windows;

internal enum AttendanceAction
{
    ClockIn,
    ClockOut
}

internal static class AttendanceActionExtensions
{
    public static string Title(this AttendanceAction action) =>
        action == AttendanceAction.ClockIn ? "출근" : "퇴근";

    public static int Code(this AttendanceAction action) =>
        action == AttendanceAction.ClockIn ? 1 : 4;

    public static string TabSelector(this AttendanceAction action) =>
        action == AttendanceAction.ClockIn
            ? "li[onclick*=\"fnSetAttOption(1)\"]"
            : "li[onclick*=\"fnSetAttOption(4)\"]";

    public static string SubmitSelector(this AttendanceAction action) =>
        action == AttendanceAction.ClockIn ? "#attHref1" : "#attHref2";

    public static string ResultSelector(this AttendanceAction action) =>
        action == AttendanceAction.ClockIn ? "#tab1" : "#tab2";
}
